### PR TITLE
Add comments to add-image-animated example

### PR DESCRIPTION
**Pull request by adrianababakanian**
Fixes https://github.com/mapbox/mapbox-gl-js-docs/issues/100

### CODE REVIEW

File: docs/pages/example/add-image-animated.html [hunk 1]
```diff
@@ -9,18 +9,21 @@
 
 var size = 200;
 
+// custom drawn pulsing dot icon
```
Comment (ryanhamley) [inline docs/pages/example/add-image-animated.html]: i think we should probably call out that this is an implementation of the CustomLayerInterface

maybe something like 

```
// implementation of CustomLayerInterface to draw a pulsing dot icon on the map
// see https://docs.mapbox.com/mapbox-gl-js/api/#customlayerinterface for more info
```
?

File: docs/pages/example/add-image-animated.html [hunk 2]
```diff
@@ -9,18 +9,21 @@
...
 var pulsingDot = {
     width: size,
     height: size,
     data: new Uint8Array(size * size * 4),
 
+    // get drawing context for icon once it is added to the map
```
Comment (ryanhamley) [inline docs/pages/example/add-image-animated.html]: `// get rendering context for the map canvas when layer is added to the map`

File: docs/pages/example/add-image-animated.html [hunk 3]
```diff
@@ -48,7 +51,7 @@
         // update this image's data with data from the canvas
         this.data = context.getImageData(0, 0, this.width, this.height).data;
 
-        // keep the map repainting
+        // keep repainting the map continuously, resulting in the smooth animation of the dot
```
Comment (ryanhamley) [inline docs/pages/example/add-image-animated.html]: could simplify this a bit to just `continuously repaint the map, resulting in the smooth animation of the dot`

File: docs/pages/example/add-image-animated.html (hunk 1 above)
Comment (ryanhamley) [reply docs/pages/example/add-image-animated.html]: or just call it a custom style layer...either way gives them more context to search the docs for

Comment (adrianababakanian) [reply docs/pages/example/add-image-animated.html]: I agree that calling out `CustomLayerInterface` and linking the relevant section in the API reference is best for docs search-ability.

**Pull request merged**